### PR TITLE
Paygroups and Roles Page

### PR DESCRIPTION
**Pull request by alainalouisepayawal**
This PR is about the CRUD operations in the Paygroups and Roles Page.

Paygroups Page
paygroups-page

Paygroups Page - Create Modal
paygroups-page-create

Paygroups Page - Update Modal
paygroups-page-update

Paygroups Page - Delete Modal
paygroups-page-delete

### CODE REVIEW

**Review by torodeveloper — CHANGES_REQUESTED** [review]
image

1. For the Shift In and Out you can use the datetime picker component
2. Disable the OK button if the form data is not yet completed.
3. Change the OK button label to "Create" and "Update" when updating.

**Review by torodeveloper — CHANGES_REQUESTED** [review]
image

Here, instead of changing the color of buttons one by one, let's add a theme to the app config and change the buttons theme there.

**Review by torodeveloper — CHANGES_REQUESTED** [review]
The Roles pagination started to display the incorrect items after navigating through pages.

**Review by torodeveloper — CHANGES_REQUESTED** [review]
image

Shouldn't this be a number input?

**Review by torodeveloper — CHANGES_REQUESTED** [review]
image

The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.

**Comment by torodeveloper** [comment]
image

What is the purpose of wrapping the core Navigation component with a custom component? Is it for the padding only?

**Comment by alainalouisepayawal** [comment]
> image
> 1. For the Shift In and Out you can use the datetime picker component
> 2. Disable the OK button if the form data is not yet completed.
> 3. Change the OK button label to "Create" and "Update" when updating.

----------------------------
1. Changed the Shift In and Shift Out input into Datetime Picker component with 12h format.
image

2. Created a function to validate data on change and will only enable the button if the data needed for the payload is complete, hence it will stay disabled.
image

image

3. Changed the button labels from “OK” to “Create” and “Update” accordingly.
image

image

**Comment by alainalouisepayawal** [comment]
> image
> Here, instead of changing the color of buttons one by one, let's add a theme to the app config and change the buttons theme there.

-------------------
1. Utilized the Theme feature of Bellini app config.
image

**Comment by alainalouisepayawal** [comment]
>  [  ](https://private-user-images.githubusercontent.com/121865290/323100882-91b251d4-68f5-443d-ab29-6bb8c88f2cb1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTAwODgyLTkxYjI1MWQ0LTY4ZjUtNDQzZC1hYjI5LTZiYjhjODhmMmNiMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWFhZTRmMGNjZDNiYThkMjE1YjZjODYxNjk2YjI0MzMzOWNiMzA3NjhiOTk2YmFkMmFmZWMzNTg0NDdiOWI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RTmPO5C2UthUu3Zz1_gH33XoodMOgpkT4KQ4BoWxgJk) [ ](https://private-user-images.githubusercontent.com/121865290/323100882-91b251d4-68f5-443d-ab29-6bb8c88f2cb1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTAwODgyLTkxYjI1MWQ0LTY4ZjUtNDQzZC1hYjI5LTZiYjhjODhmMmNiMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWFhZTRmMGNjZDNiYThkMjE1YjZjODYxNjk2YjI0MzMzOWNiMzA3NjhiOTk2YmFkMmFmZWMzNTg0NDdiOWI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RTmPO5C2UthUu3Zz1_gH33XoodMOgpkT4KQ4BoWxgJk)
> 
> The Roles pagination started to display the incorrect items after navigating through pages.

--------------------
1. Fixed the pagination function, specifically the “On Page Change.”
pagination

**Comment by alainalouisepayawal** [comment]
> image
> Shouldn't this be a number input?

-----------------
1. Changed the text input into number input for data that accepts number instead of string.
overtimeMultiplier: number             requiredHours: number
payCurrency: string                            roleTitle: string
payGroupName: string                      shiftIn: time
payRate: number                                shiftOut: time

image

**Comment by alainalouisepayawal** [comment]
> image
> The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.

------------------
1. Disabled the Update and Delete button if a row/item is not selected.
image

**Comment by alainalouisepayawal** [comment]
> image
> What is the purpose of wrapping the core Navigation component with a custom component? Is it for the padding only?

-----------------
1. Unwrapped the core navigation component from the div and added the padding using CSS.
image

**Comment by torodeveloper** [comment]
> > image
> > The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.
> 
> 1. Disabled the Update and Delete button if a row/item is not selected.
>    image


Removing the selection doesn't disable the buttons again.

File: components/new-role/stylesheets/new-role.css
Comment (torodeveloper) [inline components/new-role/stylesheets/new-role.css:1]: Is this CSS file unused?

File: stylesheets/paygroupsPage.css
Comment (torodeveloper) [inline stylesheets/paygroupsPage.css:1]: Let's stick to kebab case file naming.

Comment (torodeveloper) [reply stylesheets/paygroupsPage.css:1]: We don't have to redo it on every page. Rename this to `table-reset.css` then include all the pages to it:

```css
.paygroups-table,
.roles-table {
    text-align: center;
    overflow-y: auto;
}

etc...
```

**Review by torodeveloper — CHANGES_REQUESTED** [review]

File: components/new-role/stylesheets/new-role.css
Comment (alainalouisepayawal) [reply components/new-role/stylesheets/new-role.css:1]: Yes Sir, I forgot to delete it. I apologize for the oversight.

**Comment by alainalouisepayawal** [comment]
> > > image
> > > The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.
> > 
> > 
> > 
> > 1. Disabled the Update and Delete button if a row/item is not selected.
> >    image
> 
>  [  ](https://private-user-images.githubusercontent.com/121865290/324337301-30ffe1c5-74bf-4e53-b007-706065bcfe3f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NzAwOTcsIm5iZiI6MTcxMzc2OTc5NywicGF0aCI6Ii8xMjE4NjUyOTAvMzI0MzM3MzAxLTMwZmZlMWM1LTc0YmYtNGU1My1iMDA3LTcwNjA2NWJjZmUzZi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNzA5NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2Y0NGY1NTJmMWViNDgwNzY0MjA0M2IzMTJlNWNiMzNmYjRlZTY4Y2EwOTA3NTQ0YTU3YmZiYzFjYzFmYzEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UFDk3vMtrCHR7XlNA7ZMCcpjfc3a_cA7ZF2HU5oD5TA) [ ](https://private-user-images.githubusercontent.com/121865290/324337301-30ffe1c5-74bf-4e53-b007-706065bcfe3f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NzAwOTcsIm5iZiI6MTcxMzc2OTc5NywicGF0aCI6Ii8xMjE4NjUyOTAvMzI0MzM3MzAxLTMwZmZlMWM1LTc0YmYtNGU1My1iMDA3LTcwNjA2NWJjZmUzZi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNzA5NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2Y0NGY1NTJmMWViNDgwNzY0MjA0M2IzMTJlNWNiMzNmYjRlZTY4Y2EwOTA3NTQ0YTU3YmZiYzFjYzFmYzEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UFDk3vMtrCHR7XlNA7ZMCcpjfc3a_cA7ZF2HU5oD5TA) Removing the selection doesn't disable the buttons again.

----------------
1. Modified the selectPaygroup, selectRole and selectBenefit in order to fix the issue.

**Paygroup Page**
selectPaygroup

**Roles Page**
selectRole

**Benefits Page**
selectBenefit

File: stylesheets/paygroupsPage.css
Comment (alainalouisepayawal) [reply stylesheets/paygroupsPage.css:1]: > We don't have to redo it on every page. Rename this to `table-reset.css` then include all the pages to it:
> 
> ```css
> .paygroups-table,
> .roles-table {
>     text-align: center;
>     overflow-y: auto;
> }
> 
> etc...
> ```

----------------
1. Renamed the css to 'table-reset.css'
image

Comment (torodeveloper) [reply stylesheets/paygroupsPage.css:1]: That is not the right selector. It should be separated by a comma per page:

```css
.paygroups-table,
.roles-table,
.benifits-table { ... }

.pay-groups-table::webkit-scrollbar,
.roles-table::webkit-scrollbar,
.benifits-table::webkit-scrollbar { ... }

...
```

Comment (alainalouisepayawal) [reply stylesheets/paygroupsPage.css:1]: 1. Changed the declaration of classes in table-reset.css in order to apply the styles.
image

**Review by torodeveloper — APPROVED** [review]

**Pull request merged**